File: Simulators/gem5/nvmain_read_pmu.cc
#include "nvmain_read_pmu.hh"

#include <iostream>
#include <params/NVMainReadPMU.hh>

Nvmain_Read_PMU *Nvmain_Read_PMU::instance = 0;

Nvmain_Read_PMU::Nvmain_Read_PMU(const Params *p) : SimObject(p) {
    instance = this;
}

void Nvmain_Read_PMU::regProbePoints() {
    /**
     * Register the memory write pmu probe
     */
    ppMemBusReads.reset(new ProbePoints::PMU(getProbeManager(), "MemBusReads"));
}

void Nvmain_Read_PMU::triggerRead() {
    ppMemBusReads->notify(1);
}

Nvmain_Read_PMU *NVMainReadPMUParams::create() {
    return new Nvmain_Read_PMU(this);
}

File: Simulators/gem5/NVMainReadPMU.py
from m5.params import *
from m5.SimObject import *

class NVMainReadPMU(SimObject):
    type = 'NVMainReadPMU'
    cxx_class = 'Nvmain_Read_PMU'
    cxx_header = 'Simulators/gem5/nvmain_read_pmu.hh'

    